src/Resources/contao/widgets/FileTree.php

```diff
@@ -126,6 +168,80 @@ protected function validator($varInput)
 
 
 	/**
+	 * Check the selected value
+	 *
+	 * @param mixed $varInput
+	 */
+	protected function checkValue($varInput)
+	{
+		if ($varInput == '')
+		{
+			return;
+		}
+
+		if (strpos($varInput, ',') === false)
+		{
+			$arrUuids = array($varInput);
+		}
+		else
+		{
+			$arrUuids = array_filter(explode(',', $varInput));
+		}
+
+		$objFiles = \FilesModel::findMultipleByUuids($arrUuids);
+
+		if ($objFiles === null)
+		{
+			return;
+		}
+
+		foreach ($objFiles as $objFile)
+		{
+			// Only files can be selected
+			if ($this->filesOnly && is_dir(TL_ROOT . '/' . $objFile->path))
+			{
+				$this->addError($GLOBALS['TL_LANG']['ERR']['filesOnly']);
+				break;
+			}
+
+			// Only folders can be selected
+			if ($this->files === false && !is_dir(TL_ROOT . '/' . $objFile->path))
+			{
+				$this->addError($GLOBALS['TL_LANG']['ERR']['foldersOnly']);
+				break;
+			}
+
+			// Only files within a custom path can be selected
+			if ($this->path && strpos($objFile->path, $this->path . '/') !== 0)
+			{
+				$this->addError(sprintf($GLOBALS['TL_LANG']['ERR']['pathOnly'], $this->path));
+				break;
+			}
+
+			// Only certain file types can be selected
+			if ($this->extensions)
+			{
+				$extensions = \StringUtil::trimsplit(',', $this->extensions);
+
+				if (is_dir(TL_ROOT . '/' . $objFile->path))
+				{
+					$this->addError(sprintf($GLOBALS['TL_LANG']['ERR']['extensionsOnly'], $this->extensions));
```

Comment:
I think this case should not be an error. The `extensions` setting should only apply for files, not for directories. In a gallery element for example, you can select folders and files with image extensions. This change would break this behavior.

IMO the parent if condition should be changed to `$this->extensions && !is_dir(...)`.